### ISSUE TITLE
No-unused-vars doesn't check functions

### ISSUE THREAD

**Issue by ilyavolodin** [eslint/eslint#256]
Function declarations are marked as used automatically, because parent isn't being checked correctly. So following code will not produce an error, even though it should:

``` js
function f() {}
```
